bite-size-notes/week-6-c-state-estimation-intro.md

```diff
@@ -0,0 +1,193 @@
+
+$$
+\lim_{t\rightarrow \infty}(\tilde{x}) = 0
+$$
+
+Therefore, this now becomes a clear control problem as we're effectively trying to minimize our error.
+
+## Main Body
+
+### The Luenberger Observer
+
+The Luenberger Observer is a deterministic state estimator. First, we create create estimator copies of the original system.
+
+$$ 
+\dot{\hat{x}}(t) = Ax(t) + Bu(t) \\
+\hat{y}(t) = Cx(t) + Du(t)
+$$
+
+Now expanding our error dynamics:
+
+$$
+\dot{\tilde{x}}(t) = \dot{x}(t) - \dot{\hat{x}}(t) = Ax(t) + Bu(t) - A\hat{x}(t) - Bu(t) = A\hat{e}(t)
+$$
+
+Now we can clearly see two situations arise from the following. Either the matrix A has all eigenvalues in the left half-plane which will allow $\hat{e}(t)$ to converge to zero. While this may converge to 0, it may be impractically slow and doesn't utilize the measurement as a feedback system. Additionally, we cannot guarantee the matrix A to be stable. Adding feedback only requires adding matrix $L$ to the estimator.
+
+$$ 
+\dot{\hat{x}}(t) = Ax(t) + Bu(t) + L(y(t) - \hat{y}(t))
+$$
+
+Again we expand our error dynamics with new estimated state dynamics.
+$$
+\begin{align*}
+\dot{\tilde{x}}(t) &= \dot{x}(t) - \dot{\hat{x}}(t) \\
+&= Ax(t) + Bu(t) - A\hat{x}(t) - Bu(t) - L(y(t) - \hat{y}(t)) \\
+&= Ax(t) - A\hat{x}(t) - L(Cx(t) - C\hat{x}(t)) \\
+&= (A - LC)\tilde{x}(t)
+\end{align*}
+$$
+
+This is the equation for the Luenberger observer. As the error is now dependent on the matrix $A - LC$ which can be selected with matrix to have left hand poles. This allows the error to converge to 0.
+
+### Duality (Estimation & Control)
+
+The structure of the Luenberger Observer is very similar to state feedback dynamic equation. In a linear, time-invariant system with state-space equations and state feedback:
+
+$$
+\begin{align*}
+\dot{x} = Ax + Bu
+u = -Kx
+\end{align*}
+$$
+
+The state feedback gain matrix ($K$) can be selected to prioritize certain controls such as stability or response speed. If we use the feedback control in the original state equation, we can derive the closed-loop system:
+
+$$ 
+\dot{x} = (A-BK)x
+$$
+
+When compared to the Luenberger observer, we can see that both equations have the same form.
+
+$$
+\dot{\tilde{x}}(t) = (A^T - L^TC^T)\tilde{x}(t)
+$$
+
+To choose L and the pole placement, we need to ensure that the system must be reachable. Recalling Definition 4, we know that it is reachable when $rank(R) = n = \dim(A)$. From the duality realization, we can substitute the estimator's equivalents into our reachability matrix.
+
+$$
+\begin{align*}
+\tilde{R} &= \begin{pmatrix}
+            \tilde{B} & \tilde{A}\tilde{B} & \dots & \tilde{A}^{n-1}\tilde{B}
+            \end{pmatrix} \\
+        &= \begin{pmatrix}
+            C^T & A^T C^T & \dots & (A^T)^{n-1} C^T
+            \end{pmatrix} \\
+        &= \begin{pmatrix}
+            C \\ CA \\ \vdots \\ CA^{n-1}
+            \end{pmatrix}^T \\
+        &= O^T
+\end{align*}
+$$
+
```

Comment:
For pole placement, something might be unclear with your github code.

Reply:
Fixed the code